src/SFA.DAS.Authorization.EmployerFeatures/AuthorizationHandler.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Authorization.EmployerFeatures
+{
+    public class AuthorizationHandler : IAuthorizationHandler
+    {
+        public string Namespace => EmployerFeature.Namespace;
+        
+        private readonly IFeatureTogglesService _featureTogglesService;
+
+        public AuthorizationHandler(IFeatureTogglesService featureTogglesService)
+        {
+            _featureTogglesService = featureTogglesService;
+        }
+
+        public async Task<AuthorizationResult> GetAuthorizationResult(IReadOnlyCollection<string> options, IAuthorizationContext authorizationContext)
+        {
+            var authorizationResult = new AuthorizationResult();
+
+            if (options.Any())
+            {
+                options.EnsureNoAndOptions();
+                options.EnsureNoOrOptions();
+
+                var values = authorizationContext.GetEmployerFeatureValues();
+                var feature = options.Select(o => o.ToEnum<Feature>()).Single();
+                var featureToggle = await _featureTogglesService.GetFeatureToggle(feature).ConfigureAwait(false);
+
+                if (!featureToggle.IsEnabled)
+                {
+                    authorizationResult.AddError(new EmployerFeatureNotEnabled());
+                }
+                else if (!featureToggle.IsUserEmailWhitelisted(values.UserEmail))
```

Comment:
If the feature is enabled and there is no white listing, will the user see the feature. Looking at the code it looks like if there is no 'white listing' the feature will not be visible

Reply:
Yeah, the user should see the feature if the toggle is enabled and the user email whitelist is empty.